docs/cow-protocol/reference/core/auctions/competition_rules.md

```
@@ -18,17 +18,42 @@ All solvers participating in the solver competition must abide by certain rules.
+  - The protocol first considers bids containing only orders on the same _directed token pair_, i.e., on the same token pair and in the same direction, and computes the best bid on each directed token pair, where the best bid is the one generating the highest score. The collection of these best bids is the reference outcome for each directed token pair, representing the best possible execution against outside liquidity.
+  - The protocol then uses the reference outcome to filter out _unfair_ bids from the remaining batched bids (batched because, by definition, they contain orders on multiple directed token pairs). A batched bid is filtered out whenever it generates lower score for a given directed token pair than the reference outcome.
+  - In the final step, the protocol considers all the batched bids that survived the filtering and the best bids on individual directed token pairs. It computes the collection of winning bids, under the constraint that all orders on the same directed token pair must be part of the same winning bid.
+  Winning solvers are rewarded according to a second-price auction mechanism; for more information see the [rewards section](rewards).
+
+- Valid settlements: A settlement executed on-chain is _valid_ if:
+  - The solution was selected as winner and is executed as specified in the bidding stage with respect to solver, score, and executed amounts.
+  - Hooks of executed trades, as specified in app data of respective orders, are executed according to the following rules:
+    1. Pre-hooks need to be executed before pulling in user funds
+    2. Post-hooks need to be executed after pushing out user order proceeds
+    3. Partially fillable orders:
+      1. Should execute the pre-hooks on the first fill only
```

markdownlint-cli2 (0.18.1)

Inconsistent indentation for list items at the same level
Expected: 4; Actual: 6

(MD005, list-indent)